qiskit/circuit/instruction.py

```diff
@@ -137,29 +137,11 @@ def params(self):
     def params(self, parameters):
         self._params = []
         for single_param in parameters:
-            # example: u2(pi/2, sin(pi/4))
-            if isinstance(single_param, (ParameterExpression)):
-                self._params.append(single_param)
-            # example: u3(0.1, 0.2, 0.3)
-            elif isinstance(single_param, (int, float)):
-                self._params.append(single_param)
-            # example: Initialize([complex(0,1), complex(0,0)])
-            elif isinstance(single_param, complex):
-                self._params.append(single_param)
-            # example: snapshot('label')
-            elif isinstance(single_param, str):
-                self._params.append(single_param)
-            # example: Aer expectation_value_snapshot [complex, 'X']
-            elif isinstance(single_param, list):
-                self._params.append(single_param)
-            # example: numpy.array([[1, 0], [0, 1]])
-            elif isinstance(single_param, numpy.ndarray):
-                self._params.append(single_param)
-            elif isinstance(single_param, numpy.number):
-                self._params.append(single_param.item())
-            else:
-                raise CircuitError("invalid param type {0} in instruction "
-                                   "{1}".format(type(single_param), self.name))
+            self._params.append(self.validate_parameter(single_param))
+
+    def validate_parameter(self, parameter):
+        """Instruction parameters has no validation or normalization."""
+        return parameter
```

Comment:
Is there a reason to remove the strict type checking here as a fallback, especially for all other downstream subclasses here? This is a bit of a change in behavior and we do have subclass users of the instruction class outside of terra today. The easiest example of the top of my head is aer's snapshot instruction (which has several subclasses too): https://github.com/Qiskit/qiskit-aer/blob/master/qiskit/providers/aer/extensions/snapshot.py#L25

Reply:
As was discussed offline the conclusion to this was that it's the responsibility of users to set a sane parameter for opaque instructions or for subclasses of opaque instructions (if `validate_parameter` is not defined for the subclass)